### DISCUSSION TITLE
UNO - Windows app does not detect [NotifyPropertyChangedFor())

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#10018] [Q&A]
Is this example border background should change when IsPositive is true.

It works fine on Skia.Gtk, but it doesn't work on Windows.

I don't know if it is related to this: https://github.com/unoplatform/uno/discussions/9876

I've added a minimal example.

XAML
```
<Border x:Name="ST_Border"
        Background="Transparent"
        Height="100">

    <VisualStateManager.VisualStateGroups>
        <VisualStateGroup x:Name="Stopped">
            <VisualState x:Name="On">
                <VisualState.StateTriggers>
                    <StateTrigger IsActive="{x:Bind ViewModel.IsPositive, Mode=OneWay }" />
                </VisualState.StateTriggers>
                <VisualState.Setters>
                    <Setter Target="ST_Border.Background"
                            Value="Blue" />
                </VisualState.Setters>
            </VisualState>
        </VisualStateGroup>
    </VisualStateManager.VisualStateGroups>
```

ViewModel:
```
public partial class MainPageViewModel : ObservableObject
{
    [ObservableProperty]
    [NotifyPropertyChangedFor(nameof(IsPositive))]
    private int _stopTime = 0;

    public bool IsPositive => StopTime > 0;

    public void Increase()
    {
        StopTime += 10;
    }

    public void Decrease()
    {
        StopTime -= 10;
    }
}
```
[TestCommunityToolkit.zip](https://github.com/unoplatform/uno/files/9675785/TestCommunityToolkit.zip)

**Comment by jeromelaban** [comment]
It is likely that the generation behind `NotifyPropertyChangedFor` happens after the x:Bind has generated its support code, and it's not something we can fix (it's in Microsoft's own tooling). It's also unlikely that this will change any time soon, but it may work if you explicitly add the `INotifyPropertyChanged` interface in your code.

I would suggest to open an issue in WinUI for tracking.

**Reply by dcoartist** [reply]
Great support for Skia by the way.

**Reply by jeromelaban** [reply]
Thanks!

**Comment by mcNets** [comment]
Hi @jeromelaban.

I've tried with a WinUI project and it seems to work. The only difference between both projects is that WinUI uses a Window instead of a Page. But I'll try by using Template Studio that generates pages as well.

BTW, I opened a discussion on CommunityToolkit repo.

[TestToolkitWinUI.zip](https://github.com/unoplatform/uno/files/9693250/TestToolkitWinUI.zip)

**Reply by mcNets** [reply]
Just tested on a Template studio solution and it work too.
